app/apollo/resolvers/cluster.js

```diff
@@ -577,18 +662,18 @@ const clusterResolvers = {
       const user = whoIs(me);
 
       try {
-        logger.info({ req_id, user, org_id, registration }, `${queryName} validating`);
-
-        await validAuth(me, org_id, ACTIONS.REGISTER, TYPES.CLUSTER, queryName, context);
-
-        logger.info({ req_id, user, org_id, registration }, `${queryName} validating - authorized`);
+        logger.info({req_id, user, org_id, registration}, `${queryName} validating`);
 
         validateString( 'org_id', org_id );
         validateJson( 'registration', registration );
+        validateName( 'registration.name', registration.name );
+
         if (!registration.name) {
           throw new RazeeValidationError(context.req.t('A cluster name is not defined in the registration data'), context);
         }
-        validateName( 'registration.name', registration.name );
+
```

Comment:
Original code validated that `registration.name` exists before validating that it's value was acceptable with `validateName`.  Why reverse the order?

Reply:
fixed

Reply:
resolved